src/user_lists/models.py

```diff
@@ -0,0 +1,68 @@
+from django.db import models
+
+from researchhub_document.related_models.researchhub_unified_document_model import (
+    ResearchhubUnifiedDocument,
+)
+from utils.models import DefaultAuthenticatedModel, SoftDeletableModel
+
+
+class List(DefaultAuthenticatedModel, SoftDeletableModel):
+    name = models.CharField(max_length=120)
+    is_public = models.BooleanField(default=False)
+
+    def __str__(self):
+        return f"{self.created_by}:{self.name}"
+
+    class Meta:
+        ordering = ["name"]
+        constraints = [
+            models.UniqueConstraint(
+                fields=["created_by", "name"],
+                condition=models.Q(is_removed=False),
+                name="unique_not_removed_name_per_user",
+            )
+        ]
+        indexes = [ 
+            models.Index(
+                fields=["created_by", "is_removed"],
+                name="idx_list_user_removed",
+            ),
+        ]
+
+
+class ListItem(DefaultAuthenticatedModel, SoftDeletableModel):
+    parent_list = models.ForeignKey(
+        List,
+        on_delete=models.CASCADE,
+        related_name="items",
+    )
+
+    unified_document = models.ForeignKey(
+        ResearchhubUnifiedDocument,
+        on_delete=models.CASCADE,
+        related_name="user_list_items",
+    )
+
+    is_public = models.BooleanField(default=False)
```

Comment:
This is a level of control that is not needed. we don't need individual list items to be public/private - they should just inherit from the list.

I think it's fine to include it on parent but going forward, we  should update the model in increment when we introduce a new feature such as sharing of lists for example